pkg/skaffold/runner/dev.go

```diff
@@ -168,7 +168,7 @@ func (r *SkaffoldRunner) Dev(ctx context.Context, out io.Writer, artifacts []*la
 		default:
 			if err := r.monitor.Register(
 				func() ([]string, error) {
-					return build.DependenciesForArtifact(ctx, artifact, r.runCtx, r.artifactStore)
+					return build.DependenciesForArtifactRecursive(ctx, artifacts, artifact, r.runCtx, r.artifactStore)
```

Comment:
since you're doing the recursion here, we can skip it in the `default` case down below on line 181.
```diff
- addRebuild(g, artifact, r.changeSet.AddRebuild, r.runCtx.Opts.IsTargetImage)
+ r.changeSet.AddRebuild(artifact)
```

Reply:
ok, i understand. this should work because it will do the rebuild even if an artifact has no sync mapping at all i guess. will update.